DependencyInjection/Compiler/CacheCompatibilityPass.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Bundle\MongoDBBundle\DependencyInjection\Compiler;
+
+use Doctrine\Bundle\MongoDBBundle\DependencyInjection\DoctrineMongoDBExtension;
+use Doctrine\Common\Cache\CacheProvider;
+use Doctrine\Common\Cache\Psr6\CacheAdapter;
+use Doctrine\Common\Cache\Psr6\DoctrineProvider;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\DependencyInjection\ChildDefinition;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+use function array_keys;
+use function assert;
+use function is_a;
+use function is_string;
+use function trigger_deprecation;
+
+/** @internal */
+final class CacheCompatibilityPass implements CompilerPassInterface
+{
+    private const CACHE_SETTER_METHODS_PSR6_SUPPORT = [
+        'setMetadataCache' => true,
+        'setMetadataCacheImpl' => true,
+    ];
+
+    public function process(ContainerBuilder $container): void
+    {
+        foreach (array_keys($container->findTaggedServiceIds(DoctrineMongoDBExtension::CONFIGURATION_TAG)) as $id) {
+            /** @var array<int, string|Reference[]> $methodCall */
+            foreach ($container->getDefinition($id)->getMethodCalls() as $methodCall) {
```

Comment:
Really not sure why the dependency-injection component doesn't have a `MethodCall` class.. but basically spent way too long figuring out that `$methodCall` is an array where index 0 points to the method name as a string, and index 1 points to an array of references which represent the arguments of the methodcall. Figured some additional phpdocs wouldn't hurt and neither would the assert call made here, but might be wrong about that :)

Reply:
You might want to file an issue for this or reach out to some of the core Symfony folks about this. I'm not sure how that could be introduced without BC breaks, but I do agree that using arrays for this stuff seems weird. Maybe @nicolas-grekas can provide some insight into this.

Reply:
In the meantime we can use an array-shape like:

```php
/** @var array{0: string, 1: mixed[]} $methodCall */
```
based on [`addMethodCall`](https://github.com/symfony/symfony/blob/76a414c7d8ad03b5d6e520edfd8b947af21717b0/src/Symfony/Component/DependencyInjection/Definition.php#L353) I think it's not only `Reference[]`, it could be any argument (scalar, arrays, etc).

This way the assert won't be necessary.

I don't know if the Symfony people would accept array-shapes in the phpdoc, that would be quite useful.

Reply:
That's a great suggestion, and good point. Didn't consider it could be something than `Reference[]`